corehq/apps/users/views/utils.py

```diff
@@ -113,26 +115,58 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
         )
 
 
-def get_locations_with_cases(domain, location_ids):
-    """
-    Takes a list of location IDs and returns how many cases are assigned to each location
-    :param domain: The domain to search in
-    :param location_ids: A list of location IDs to get case counts on
-    :returns A dict with location names as the key, and the number of cases assigned to it as the value
-    """
+def _get_location_ids_with_single_user(domain, location_ids, user_id):
+    # Remove the user's ID, as we want to see which other CommCare users share any of the given location_ids
+    user_query = (
+        UserES()
+        .domain(domain)
+        .mobile_users()
+        .location(location_ids)
+    )
+    user_query.filter(filters.NOT(filters.doc_id(user_id)))
+    other_commcare_user_ids = user_query.get_ids()
+
+    case_query = (
+        CaseES()
+        .domain(domain)
+        .owner(location_ids)
+        .user(other_commcare_user_ids)
+        .aggregation(TermsAggregation('by_location', 'owner_id')
+        .size(0))
+    ).run()
+    return list(set(location_ids) - set(case_query.aggregations.by_location.keys))
+
+
+def _get_location_case_counts_with_single_user(domain, location_ids):
     query = (CaseES()
             .domain(domain)
             .owner(location_ids)
             .aggregation(TermsAggregation('by_location', 'owner_id')
-            .size(0)))
-    counts = query.run().aggregations.by_location.counts_by_bucket()
-    locations_with_cases = dict()
-    if counts:
-        locations = SQLLocation.objects.filter(location_id__in=location_ids)
+            .size(0))
+    ).run()
+    locations = SQLLocation.objects.filter(location_id__in=location_ids)
+    counts = query.aggregations.by_location.counts_by_bucket()
+    return locations, counts
+
+
+def get_locations_with_single_user(domain, location_ids, user_id):
```

Comment:
I'm not sure this query is correct as written.  The case query queries on `.user()`, which is the last user to have updated the case.  That's can also be things like web users who did an upload, or "system" for system actions.  I think case ownership is the only thing that matters.  I was having trouble wrapping my head around the set of queries here, so I decided to take a crack at it myself.  Here's my attempt (though I haven't tested it):

```python
def get_locations_with_orphaned_cases(domain, location_ids, user_id):
    # user_id is currently assigned to each location in this set
    counts_per_loc = _get_case_counts_per_location(domain, location_ids)
    locs_with_other_users = _get_locations_with_other_users(domain, location_ids, user_id)
    return {
        location_id: count for location_id, count in counts_per_loc.items()
        if location_id not in locs_with_other_users
    }


def _get_case_counts_per_location(domain, location_ids):
    query = (CaseES()
             .domain(domain)
             .owner(location_ids)
             .aggregation(TermsAggregation('by_location', 'owner_id').size(0)))
    query.run().aggregations.by_location.counts_by_bucket()


def _get_locations_with_other_users(domain, location_ids, user_id):
    loc_ids_per_user = (UserES()
                        .domain(domain)
                        .mobile_users()
                        .NOT(filters.doc_id(user_id))
                        .location(location_ids)
                        .values_list('assigned_location_ids', flat=True))
    # Each of the locations in these lists must have another user
    return set(chain(*loc_ids_per_user))
```

Reply:
Thanks for pointing that out. The query could indeed possibly give incorrect results as it would only return locations where the other users were the last person to modify a case, not necessarily create it. Thanks for the example code, I'll be sure to remove the unnecessary query filter. :+1: